### ISSUE TITLE
Block Gap for the inserter is getting ignored on last elements

### ISSUE THREAD

**Issue by hanneslsm** [WordPress/gutenberg#55433]
### Description

When I am having a column block and some groups within, I'd like to move the groups around. The column has a block gap, which isn't displayed on the last element when drag & dropping the groups. The inserter should have a padding (= half the block gap).

### Step-by-step reproduction instructions

1. Drag & Drop a block element in a area where a block gap is defined.
2. Drag & Drop the block to the last element. Watch the block gap.

### Screenshots, screen recording, code snippet

https://github.com/WordPress/gutenberg/assets/17861443/99865f8e-b148-4990-9cb6-1e24728ad8f5

### Environment info

- WP  6.3.2 
- GB 16.8.1
- Custom theme

### Please confirm that you have searched existing issues in the repo.

No

### Please confirm that you have tested with all plugins deactivated except Gutenberg.

No

**Comment by jordesign** [comment]
Confirmed this in testing with WP 6.3.2 and TT3

**Comment by ramonjd** [comment]
Just to confirm, is the expectation that the inbetween inserter line will reflect the block gap value that _will_ take effect after dropping?

If so, it's a tricky one. The inbetween inserter popup is rendered above the block list (under `.edit-post-visual-editor`) so it's not really context aware.

I had a quick look to see if we could apply some bottom margin to the last-child of a container that has block gap applied when the inserter is being dragged, but couldn't get it to work.

The inbetween inserter has been explicitly given a zero margin:

https://github.com/WordPress/gutenberg/blob/f1e4c2ad3469a0aa4b8edfd757f17e03309a29f2/packages/block-editor/src/components/block-popover/style.scss#L7-L7

So the idea is that, if a block container with layout knows that the block inserter is positioned after the last-child we add the relevant blockGap value to the inserter's margin. 

But in practice it's not straight forward.

**Comment by hanneslsm** [comment]
> Just to confirm, is the expectation that the inbetween inserter line will reflect the block gap value that will take effect after dropping?

Not sure if I got this.
Let me rephrase to avoid misunderstandings: The expectation is that via the inserter (blue line) represents already if there is a block gap or not. This works perfectly fine in between elements. But when dropping it below the last element, the inserter tells me that there won't be a gap… but after dropping it "suddenly appears".

I see how this is becoming tricky, especially when considering horizontal blocks. 
However, I think that the is inserter is in its core a "preview" on where the blocks can be dropped, so we should aim for the best preview possible.

**Comment by ramonjd** [comment]
> The expectation is that via the inserter (blue line) represents already if there is a block gap or not.

Thanks for the clarification. It's what I understood, but I phrased it poorly, sorry! 👍🏻 

I agree with your sentiments in relation to getting the preview state right while dragging, I'm just not sure about the best way to reflect this yet. 🤔
